hipogram/posts/admin.py

```diff
@@ -1,17 +1,49 @@
 from django.contrib import admin
 from .models import Post, Tag, Like, Rate
+from django.contrib.auth.models import User
 
-admin.site.register(Tag)
-admin.site.register(Like)
-admin.site.register(Rate)
+
+class LikeInline(admin.TabularInline):
+    model = Like
+
+
+class RateInline(admin.TabularInline):
+    model = Rate
+
+
+class PostInline(admin.TabularInline):
+    model = Post
 
 
 class PostAdmin(admin.ModelAdmin):
     list_display = ("created_by", "creation_datetime", "_tags")
     list_filter = ("created_by", "tags")
+    inlines = [LikeInline, RateInline]
 
     def _tags(self, post):
         return list(post.tags.all())
 
 
+class LikeAdmin(admin.ModelAdmin):
+    list_filter = ("post", "user")
+    list_display = ("post", "user")
+    readonly_fields = ("post", "user")
+
+
+class RateAdmin(admin.ModelAdmin):
+    list_filter = ("post", "user")
+    list_display = ("post", "user", "value")
+    readonly_fields = ("post", "user", "value")
+
+
+class UserAdmin(admin.ModelAdmin):
+    list_display = ("username", "first_name", "email", "is_active", "is_staff")
+    inlines = [PostInline]
+
+
 admin.site.register(Post, PostAdmin)
+admin.site.register(Like, LikeAdmin)
```

Comment:
FYI: there is a decorator for registering admin sections which is better to use IMO: https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#the-register-decorator